### ISSUE TITLE
HTML and Reveal.js export broken since Jinja2 3.1.x

### ISSUE THREAD

**Issue by kathoef** [jupyter/nbconvert#1739]
Take any Jupyter notebook and try to convert it,

```
$ jupyter nbconvert --to=html Untitled.ipynb 
[NbConvertApp] Converting notebook Untitled.ipynb to html
Traceback (most recent call last):
  File "/usr/local/bin/jupyter-nbconvert", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/jupyter_core/application.py", line 264, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/traitlets/config/application.py", line 846, in launch_instance
    app.start()
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 369, in start
    self.convert_notebooks()
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 541, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 506, in convert_single_notebook
    output, resources = self.export_single_notebook(notebook_filename, resources, input_buffer=input_buffer)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 435, in export_single_notebook
    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/exporter.py", line 190, in from_filename
    return self.from_file(f, resources=resources, **kw)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/exporter.py", line 208, in from_file
    return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/html.py", line 210, in from_notebook_node
    return super().from_notebook_node(nb, resources, **kw)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/templateexporter.py", line 392, in from_notebook_node
    output = self.template.render(nb=nb_copy, resources=resources)
  File "/usr/local/lib/python3.9/dist-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.9/dist-packages/jinja2/environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 3, in top-level template code
    {% from 'jupyter_widgets.html.j2' import jupyter_widgets %}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/base.html.j2", line 260, in top-level template code
    {% set div_id = uuid4() %}
  File "/usr/local/share/jupyter/nbconvert/templates/base/display_priority.j2", line 1, in top-level template code
    {%- extends 'base/null.j2' -%}
  File "/usr/local/share/jupyter/nbconvert/templates/base/null.j2", line 24, in top-level template code
    {%- block header -%}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 9, in block 'header'
    {%- block html_head -%}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 36, in block 'html_head'
    {% block notebook_css %}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 37, in block 'notebook_css'
    {{ resources.include_css("static/index.css") }}
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/html.py", line 216, in resources_include_css
    return jinja2.Markup(code)
AttributeError: module 'jinja2' has no attribute 'Markup'
```
The same problem applies to `jupyter nbconvert --to slides Untitled.ipynb`.

Updating the Jinja2 package version does not help,

```
$ pip install Jinja2==3.1.1
Collecting Jinja2==3.1.1
  Downloading Jinja2-3.1.1-py3-none-any.whl (132 kB)
     |████████████████████████████████| 132 kB 3.4 MB/s 
Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.9/dist-packages (from Jinja2==3.1.1) (2.1.1)
Installing collected packages: Jinja2
  Attempting uninstall: Jinja2
    Found existing installation: Jinja2 3.1.0
    Uninstalling Jinja2-3.1.0:
      Successfully uninstalled Jinja2-3.1.0
Successfully installed Jinja2-3.1.1
$ jupyter nbconvert --to=html Untitled.ipynb 
[NbConvertApp] Converting notebook Untitled.ipynb to html
Traceback (most recent call last):
  File "/usr/local/bin/jupyter-nbconvert", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/jupyter_core/application.py", line 264, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/traitlets/config/application.py", line 846, in launch_instance
    app.start()
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 369, in start
    self.convert_notebooks()
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 541, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 506, in convert_single_notebook
    output, resources = self.export_single_notebook(notebook_filename, resources, input_buffer=input_buffer)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/nbconvertapp.py", line 435, in export_single_notebook
    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/exporter.py", line 190, in from_filename
    return self.from_file(f, resources=resources, **kw)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/exporter.py", line 208, in from_file
    return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/html.py", line 210, in from_notebook_node
    return super().from_notebook_node(nb, resources, **kw)
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/templateexporter.py", line 392, in from_notebook_node
    output = self.template.render(nb=nb_copy, resources=resources)
  File "/usr/local/lib/python3.9/dist-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.9/dist-packages/jinja2/environment.py", line 926, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 3, in top-level template code
    {% from 'jupyter_widgets.html.j2' import jupyter_widgets %}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/base.html.j2", line 260, in top-level template code
    {% set div_id = uuid4() %}
  File "/usr/local/share/jupyter/nbconvert/templates/base/display_priority.j2", line 1, in top-level template code
    {%- extends 'base/null.j2' -%}
  File "/usr/local/share/jupyter/nbconvert/templates/base/null.j2", line 24, in top-level template code
    {%- block header -%}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 9, in block 'header'
    {%- block html_head -%}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 36, in block 'html_head'
    {% block notebook_css %}
  File "/usr/local/share/jupyter/nbconvert/templates/lab/index.html.j2", line 37, in block 'notebook_css'
    {{ resources.include_css("static/index.css") }}
  File "/usr/local/lib/python3.9/dist-packages/nbconvert/exporters/html.py", line 216, in resources_include_css
    return jinja2.Markup(code)
AttributeError: module 'jinja2' has no attribute 'Markup'
root@4370acf06cd6:/home/jovyan# 
```

But installing an older Jinja2 version does,

```
$ pip install Jinja2==3.0.*
Collecting Jinja2==3.0.*
  Downloading Jinja2-3.0.3-py3-none-any.whl (133 kB)
     |████████████████████████████████| 133 kB 3.3 MB/s 
Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.9/dist-packages (from Jinja2==3.0.*) (2.1.1)
Installing collected packages: Jinja2
  Attempting uninstall: Jinja2
    Found existing installation: Jinja2 3.1.1
    Uninstalling Jinja2-3.1.1:
      Successfully uninstalled Jinja2-3.1.1
Successfully installed Jinja2-3.0.3
root@4370acf06cd6:/home/jovyan# jupyter nbconvert --to=html Untitled.ipynb 
[NbConvertApp] Converting notebook Untitled.ipynb to html
[NbConvertApp] Writing 574036 bytes to Untitled.html
```

These are my original package versions,

```
$ pip list
Package              Version
-------------------- ---------
alembic              1.7.7
anyio                3.5.0
argon2-cffi          21.3.0
argon2-cffi-bindings 21.2.0
asttokens            2.0.5
async-generator      1.10
attrs                21.4.0
Babel                2.9.1
backcall             0.2.0
beautifulsoup4       4.10.0
bleach               4.1.0
certifi              2021.10.8
certipy              0.1.3
cffi                 1.15.0
charset-normalizer   2.0.12
cryptography         36.0.2
debugpy              1.6.0
decorator            5.1.1
defusedxml           0.7.1
entrypoints          0.4
executing            0.8.3
greenlet             1.1.2
gyp                  0.1
idna                 3.3
ipykernel            6.9.2
ipython              8.1.1
ipython-genutils     0.2.0
jedi                 0.18.1
Jinja2               3.1.0
json5                0.9.6
jsonschema           4.4.0
jupyter-client       7.1.2
jupyter-core         4.9.2
jupyter-server       1.15.6
jupyter-telemetry    0.1.0
jupyterhub           1.5.0
jupyterlab           3.3.2
jupyterlab-pygments  0.1.2
jupyterlab-server    2.11.2
Mako                 1.2.0
MarkupSafe           2.1.1
matplotlib-inline    0.1.3
mistune              0.8.4
nbclassic            0.3.7
nbclient             0.5.13
nbconvert            6.4.4
nbformat             5.2.0
nest-asyncio         1.5.4
notebook             6.4.8
notebook-shim        0.1.0
oauthlib             3.2.0
packaging            21.3
pamela               1.0.0
pandocfilters        1.5.0
parso                0.8.3
pexpect              4.8.0
pickleshare          0.7.5
pip                  20.3.4
prometheus-client    0.13.1
prompt-toolkit       3.0.28
psutil               5.9.0
ptyprocess           0.7.0
pure-eval            0.2.2
pycparser            2.21
Pygments             2.11.2
pyOpenSSL            22.0.0
pyparsing            3.0.7
pyrsistent           0.18.1
python-dateutil      2.8.2
python-json-logger   2.0.2
pytz                 2022.1
pyzmq                22.3.0
requests             2.27.1
ruamel.yaml          0.17.21
ruamel.yaml.clib     0.2.6
Send2Trash           1.8.0
setuptools           52.0.0
six                  1.16.0
sniffio              1.2.0
soupsieve            2.3.1
SQLAlchemy           1.4.32
stack-data           0.2.0
terminado            0.13.3
testpath             0.6.0
tornado              6.1
traitlets            5.1.1
urllib3              1.26.9
wcwidth              0.2.5
webencodings         0.5.1
websocket-client     1.3.1
wheel                0.34.2
```

I will use the older Jinja2 version in the meantime, but solving this would be highly appreciated, as a fully functional HTML export is an important feature in our Jupyter for teaching setup.

**Comment by minrk** [comment]
I believe this is fixed in nbconvert 6.4.5, just out

**Comment by kathoef** [comment]
Thank you, it works! 🎉
